File: web/src/api/scheduled_task.ts
// 定时任务配置
import apiClient from "@/api/client.ts";

export type LastRunStatus = 'unknown' | 'success' | 'failed' | 'ambiguous';

export interface ScheduledTask {
    id: string;
    name: string;
    enabled: boolean;
    intervalDays: number;
    phoneNumber: string;
    content: string;
    simId: string;
    // 仅用于识别尚未迁移的旧任务，新的保存请求不会再提交 deviceId。
    deviceId?: string;
    createdAt?: number;
    lastRunAt?: number;
    lastMsgId?: string;
    lastRunStatus?: LastRunStatus;
}

export type ScheduledTaskInput = Pick<
    ScheduledTask,
    'simId' | 'name' | 'enabled' | 'intervalDays' | 'phoneNumber' | 'content'
>;

export interface ScheduledTaskTriggerResponse {
	message: string;
	messageId: string;
	status?: 'ambiguous';
}

export interface ScheduledTaskTriggerRequest {
	id: string;
	requestId: string;
}

// 定时任务 API (RESTful)
// 获取所有定时任务
export const getScheduledTasks = () => {
    return apiClient.get<ScheduledTask[]>('/scheduled-tasks');
};

// 获取单个定时任务
export const getScheduledTask = (id: string) => {
    return apiClient.get<ScheduledTask>(`/scheduled-tasks/${id}`);
};

// 创建定时任务
export const createScheduledTask = (task: ScheduledTaskInput) => {
    return apiClient.post<ScheduledTask>('/scheduled-tasks', task);
};

// 更新定时任务
export const updateScheduledTask = (id: string, task: ScheduledTaskInput) => {
    return apiClient.put<ScheduledTask>(`/scheduled-tasks/${id}`, task);
};

// 删除定时任务
export const deleteScheduledTask = (id: string) => {
    return apiClient.delete<{ message: string }>(`/scheduled-tasks/${id}`);
};

// 立即触发定时任务
export const triggerScheduledTask = ({id, requestId}: ScheduledTaskTriggerRequest) => {
	return apiClient.post<ScheduledTaskTriggerResponse>(
		`/scheduled-tasks/${id}/trigger`,
		{requestId},
		{headers: {'Idempotency-Key': requestId}},
	);
};

File: internal/models/scheduled_task.go
package models

type LastRunStatus string

const (
	LastRunStatusUnknown   LastRunStatus = "unknown"
	LastRunStatusSuccess   LastRunStatus = "success"
	LastRunStatusFailed    LastRunStatus = "failed"
	LastRunStatusAmbiguous LastRunStatus = "ambiguous"
)

// ScheduledTask 定时任务
type ScheduledTask struct {
	ID           string `gorm:"primaryKey" json:"id"`                  // UUID
	DeviceID     string `gorm:"index" json:"deviceId,omitempty"`       // 旧版物理设备字段，仅保留审计/迁移，禁止作为发送回退
	SIMID        string `gorm:"index" json:"simId"`                    // 目标 SIM（由 ICCID 派生）；发送时必须填写
	Name         string `json:"name"`                                  // 任务名称
	Enabled      bool   `json:"enabled"`                               // 是否启用
	IntervalDays int    `json:"intervalDays"`                          // 执行间隔天数，例如 90 表示每90天执行一次
	PhoneNumber  string `json:"phoneNumber"`                           // 目标手机号
	Content      string `gorm:"type:text" json:"content"`              // 短信内容
	CreatedAt    int64  `json:"createdAt" gorm:"autoCreateTime:milli"` // 创建时间（时间戳毫秒）
	UpdatedAt    int64  `json:"updatedAt" gorm:"autoUpdateTime:milli"` // 更新时间（时间戳毫秒）

	LastMsgId     string        `json:"lastMsgId"`     // 上次发送的短信ID
	LastRunAt     int64         `json:"lastRunAt"`     // 上次执行时间（时间戳毫秒）
	LastRunStatus LastRunStatus `json:"lastRunStatus"` // 上次执行状态
}

func (ScheduledTask) TableName() string {
	return "scheduled_tasks"
}
